src/components/screens/forms/forms-screen.tsx

```diff
@@ -9,105 +9,173 @@ import NotificationsActiveIcon from "@mui/icons-material/NotificationsActive";
-    headerName: strings.navigationHeader.formsScreens.formScreen.form.newReply,
-    width: 250,
-    renderHeader: params => {
-      return (
-        <AdminFormListStack direction="row">
-          <NotificationsIcon style={ { fill: "darkgrey" } }/>
-          <AdminFormTypographyField sx={{ fontWeight: "bold" }}>{ params.colDef.headerName }</AdminFormTypographyField>
-        </AdminFormListStack>
-      );
-    },
-    renderCell: params => {
-      return (
-        <AdminFormListStack direction="row">
-          <NotificationsActiveIcon style={ { fill: "red" } }/>
-          <AdminFormTypographyField>{ params.row.newReply }</AdminFormTypographyField>
-        </AdminFormListStack>
-      );
+  ];
+
+  const [ rows, setRows ] = useState<Row[]>([]);
+  const [ loading, setLoading ] = useState(false);
+
+  /**
+   * Gets the latest reply date of a Metaform
+   */
+  const getLatestReplyDate = (replies: Reply[]) => {
+    if (replies.length < 1) {
+      return "";
     }
-  }
-];
-
-const rows = [
-  {
-    id: "SOSTERI-Huoli-vanhuksesta", latestReply: "01.0.1.2022 16:48", newReply: strings.navigationHeader.formsScreens.formScreen.form.notProcessed
-  },
-  {
-    id: "ESSOTE-Huoli-vanhuksesta", latestReply: "01.0.1.2022 16:48", newReply: "-"
-  }
-];
 
-/**
- * Forms screen component
- */
-const FormsScreen: React.FC = () => (
-  <>
-    <NavigationTabContainer>
-      <NavigationTab
-        text={ strings.navigationHeader.formsScreens.formScreen }
-      />
-      <NavigationTab
-        text={ strings.navigationHeader.formsScreens.formDataScreen }
-        to="data"
+    if (!replies[0].modifiedAt) {
+      return "";
+    }
+    
+    return replies[0].modifiedAt.toLocaleString().slice(0, -3);
+  };
+
+  /**
+   * Counts amount of waiting replies to be displayed in the row
+   * @param replies
+   */
+  const countWaitingReplies = (replies: Reply[]) => replies.filter(reply => reply.data?.status as (string | undefined) === "waiting").length;
+
+  /**
+   * Builds a row for the table
+   * 
+   * @param form form
+   * @param replies replies
+   */
+  const buildRow = (form: Metaform, replies: Reply[]) => {
+    const amountWaiting = countWaitingReplies(replies);
+
+    return {
+      id: form.title || strings.formScreen.noTitle,
+      latestReply: getLatestReplyDate(replies),
+      newReply: amountWaiting > 0 ? `${strings.navigationHeader.formsScreens.formScreen.form.notProcessed} (${amountWaiting})` : undefined
+    };
+  };
+
+  /**
+   * View setup
+   */
+  const setup = async () => {
+    setLoading(true);
+
+    try {
+      const forms = await metaformsApi.listMetaforms({});
+      const replies = await Promise.all(forms.map(form => repliesApi.listReplies({ metaformId: form.id!! })));
+      const builtRows = forms.map((form, i) => buildRow(form, replies[i]));
+
+      setRows(builtRows);
+    // eslint-disable-next-line no-empty
+    } catch (e) {}
+
+    setLoading(false);
+  };
+
+  useEffect(() => {
+    setup();
+  }, []);
```

Comment:
I think setup method should be at the start of the component for better readability.